### ISSUE TITLE
`npm cache add <tarball>` is not working as expected

### ISSUE THREAD

**Issue by ORESoftware** [npm/npm#20616]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [x] npm is doing something I don't understand.
  - [ ] npm is producing incorrect or undesirable behavior.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?

I have this question on SO:
https://stackoverflow.com/questions/50339371/npm-cache-add-tgz-does-not-work

basically I am doing this, I pack an NPM library with:

```bash
tarball="$(npm pack)";
```

then I run

```bash
npm cache add "$PWD/$tarball"
```

but when I inspect the npm cache in .npm, there's nothing new in there.

Does anyone know why the NPM cache would not get updated by the add command I ran?

**Comment by legodude17** [comment]
Try running `npm pack --silent` instead.
